core/src/main/java/com/microsoft/applicationinsights/internal/authentication/AadAuthentication.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    public static AadAuthentication getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("must init before using");
+        }
+        return instance;
+    }
+
+    private AadAuthentication(AuthenticationType authenticationType, String clientId, String keePassDatabasePath,
+                             String tenantId, String clientSecret, String authorityHost) {
+        this.authenticationType = authenticationType;
+        this.clientId = clientId;
+        this.keePassDatabasePath = keePassDatabasePath;
+        this.tenantId = tenantId;
+        this.clientSecret = clientSecret;
+        this.authorityHost = authorityHost;
+    }
+
+    public HttpPipelinePolicy getAuthenticationPolicy() {
+        if (authenticationType == null) return null;
+        switch (authenticationType) {
+            case UAMI:
+                return getAuthenticationPolicyWithUAMI();
+            case INTELLIJ:
+                return getAuthenticationPolicyWithIntellij();
+            case SAMI:
+                return getAuthenticationPolicyWithSAMI();
+            case VSCODE:
+                return getAuthenticationPolicyWithVsCode();
+            case CLIENTSECRET:
+                return getAuthenticationPolicyWithClientSecret();
+            default:
+                throw new IllegalStateException("Invalid Authentication Type used in AAD Authentication");
+        }
+    }
+
+    private HttpPipelinePolicy getAuthenticationPolicyWithIntellij() {
+        IntelliJCredentialBuilder intelliJCredential = new IntelliJCredentialBuilder();
+        // KeePass configuration required only for Windows. No configuration needed for Linux / Mac
+        if(SystemInformation.INSTANCE.isWindows() && keePassDatabasePath != null) {
+            intelliJCredential.keePassDatabasePath(keePassDatabasePath);
+        }
+        return new BearerTokenAuthenticationPolicy(intelliJCredential.build(), APPLICATIONINSIGHTS_AUTHENTICATION_SCOPE);
+    }
+
+    private HttpPipelinePolicy getAuthenticationPolicyWithUAMI() {
+        ManagedIdentityCredentialBuilder managedIdentityCredential = new ManagedIdentityCredentialBuilder();
+        if(clientId != null) {
+            managedIdentityCredential.clientId(clientId);
+        }
+        return new BearerTokenAuthenticationPolicy(managedIdentityCredential.build(), APPLICATIONINSIGHTS_AUTHENTICATION_SCOPE);
+    }
+
+    private HttpPipelinePolicy getAuthenticationPolicyWithClientSecret() {
+        ClientSecretCredentialBuilder credential = new ClientSecretCredentialBuilder();
+        if(tenantId != null) {
+            credential.tenantId(tenantId);
+        }
+        if(clientSecret != null) {
+            credential.clientSecret(clientSecret);
+        }
+        if(clientId != null) {
+            credential.clientId(clientId);
+        }
+        if (authorityHost != null) {
+            credential.authorityHost(authorityHost);
+        }
```

Comment:
are all these null checks needed? is it harmful to pass null to these builder methods?

Reply:
some of the builder methods are validating and throwing exceptions if its not a valid input. I added these to avoid these exceptions.  For example validateAuthHost in ClientSecretCredentialBuilder

Reply:
I sent a commit to your branch to remove a couple of these that do not look needed